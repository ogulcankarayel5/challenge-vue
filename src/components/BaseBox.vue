<template>
  <div class="box">
    <slot />
  </div>
</template>

<script>
export default {
  name: "BaseBox",
};
</script>

<style scoped lang="sass">
.box
  background-color: #E8E8E8
  width: 100px
  height: 100px
  justify-content: center
  align-items: center
  flex-direction: column
  display: flex
  border: 1px solid   #606060
  border-radius: 7px
</style>
