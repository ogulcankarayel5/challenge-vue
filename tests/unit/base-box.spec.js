import { mount } from "@vue/test-utils";
import BaseBox from "@/components/BaseBox.vue";

describe("BaseBox.vue", () => {
  it("snapshot", () => {
    const wrapper = mount(BaseBox, {
      slots: {
        default: "<div>basebox</div>",
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
  it("renders component correctly with slot", () => {
    const wrapper = mount(BaseBox, {
      slots: {
        default: "<div>basebox</div>",
      },
    });
    expect(wrapper.html()).toContain("<div>basebox</div>");
  });

  it("renders component with correct class", () => {
    const wrapper = mount(BaseBox);
    expect(wrapper.find(".box")).toBeTruthy();
  });
});
