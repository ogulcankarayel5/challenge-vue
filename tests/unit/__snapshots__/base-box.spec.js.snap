// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseBox.vue snapshot 1`] = `
<div
  class="box"
>
  
  
  <div>
    basebox
  </div>
  
  
</div>
`;
